### DISCUSSION TITLE
Example of extending authenticatation logic

### DISCUSSION THREAD

**Discussion by DmiPet** [keystonejs/keystone#6556] [Event Q&A Q3 2021]
Hi team Keystone! Would love to see/discuss ways of extending the authentication functionality with custom logic, 3rd party auth providers, etc.

**Comment by raveling** [comment]
@DmiPet thanks for your question 🙌 

~We answered this in the recent community Q&A event:
https://www.youtube.com/watch?v=r1IJh-iHm1c&t=1462s~

edit: see https://github.com/keystonejs/keystone/discussions/6576#discussioncomment-2800697

**Reply by biapar** [reply]
No video.
